ui/src/containers/Details/index.tsx

```diff
@@ -6,16 +6,28 @@ import { useMst } from '../../store/root';
 import BasicDetails from '../BasicDetails';
 import Description from '../../components/Description';
 import { assert } from '../../store/utils';
+import { NotFound } from '../../components/PageNotFound';
+import { titleCase } from '../../common/titlecase';
+import { ICatalog } from '../../store/catalog';
 
 const Details: React.FC = () => {
   const { resources, user } = useMst();
-  const { name } = useParams();
+  const { name, catalog, kind } = useParams();
+
+  const validateUrl = () => {
+    let catalogUrl: boolean | undefined;
+    resources.catalogs.values.forEach((item: ICatalog) => {
```

Comment:
Can we rename this to first 
`const catalog = resources.catalogs.values` before `validateUrl` function and then use here as `catalog.forEach`